packages/flutter/test/material/text_field_test.dart

```diff
@@ -191,6 +191,39 @@ void main() {
     await tester.pump();
   });
 
+  testWidgets('Caret position is updated on tap', (WidgetTester tester) async {
+    final TextEditingController controller = new TextEditingController();
+
+    Widget builder() {
+      return overlay(new Center(
+        child: new Material(
+          child: new TextField(
+            controller: controller,
+          ),
+        ),
+      ));
+    }
+
+    await tester.pumpWidget(builder());
+    expect(controller.selection.baseOffset, -1);
+    expect(controller.selection.extentOffset, -1);
+
+    final String testValue = 'abc def ghi';
+    await tester.enterText(find.byType(EditableText), testValue);
+    await tester.idle();
```

Comment:
This idle shouldn't be needed.  It used to be needed, but at HEAD it is folded into enterText